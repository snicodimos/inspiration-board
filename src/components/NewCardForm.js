import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

// const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {

  constructor(props) {
    super(props);

    this.state = {
      text: "",
      emoji: "",
    }
  }
  onInputChange = (event) => {
    console.log("Hit the onInputChange funtion");

    const field = event.target.name;
    const value = event.target.value;

    const newState = {};
    newState[field] = value;
    this.setState(newState);
  }

  onFormSubmit = (event) => {
    event.preventDafault();

    const newCard = {
      text: this.state.text,
      emoji: this.state.emoji,
    };

    this.setState({
      text: '',
      emoji: '',
    });
    console.log("created a new card from input", newCard);
    this.props.addNewCardToCollection(newCard);
  }

  render() {
    return (
      <div className="new-card-form">
        <div className="new-card-form__header">
          <h2>New card Submission Form</h2>
        </div>
        <form
          className="new-card-form__form"
          onSubmit={this.onFormSubmit}>
            <label for="text"> Message</label>
            <input
              className="new-card-form__form-textarea"
              type="text"
              name="text"
              value={this.state.text}
              onChange={this.onInputChange}
              />

            Emoji

            <input
              className="new-card-form__form-select"
              placeholder = "heart_eyes"
              type="text"
              name="emoji"
              value={this.state.emoji}
              onChange={this.onInputChange}
              />
            <input
              type="submit"
              value="Submit Message"
              className="new-card-form__form-button"
              />
        </form>
      </div>
    )
  }
};

export default NewCardForm;
