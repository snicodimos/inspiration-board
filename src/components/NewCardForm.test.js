import React from 'react';
import NewCardForm from './NewCardForm';
import { shallow } from 'enzyme';


describe('NewCardForm', () => {
  test('that it matches an existing snapshot for adding', () => {

    const wrapper = shallow( <NewCardForm addNewCardToCollection={() => {} } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('that it matches an existing snapshot for delete', () => {

    const wrapper = shallow( <NewCardForm deleteCardCallback={() => {} } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('that it matches an existing snapshot for delete', () => {

    const wrapper = shallow( <NewCardForm deleteCardCallback={() => {} } />);

    expect(wrapper).toMatchSnapshot();
  });
});
