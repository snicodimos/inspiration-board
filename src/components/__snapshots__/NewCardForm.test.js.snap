// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot for adding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    addNewCardToCollection={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="new-card-form__header"
        >
          <h2>
            New card Submission Form
          </h2>
        </div>,
        <form
          className="new-card-form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="text"
          >
            Your Message
          </label>
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            onChange={[Function]}
            value=""
          />
          <label
            htmlFor="emoji"
          >
             Emoji
          </label>
          <input
            className="new-card-form__form-select"
            name="emoji"
            onChange={[Function]}
            value=""
          />
          <input
            className="new-card-form__form-button"
            type="submit"
            value="Submit Message"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            New card Submission Form
          </h2>,
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New card Submission Form",
          },
          "ref": null,
          "rendered": "New card Submission Form",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="text"
            >
              Your Message
            </label>,
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />,
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>,
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />,
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />,
          ],
          "className": "new-card-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Message",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Your Message",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "text",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Emoji",
              "htmlFor": "emoji",
            },
            "ref": null,
            "rendered": " Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-button",
              "type": "submit",
              "value": "Submit Message",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="new-card-form__header"
          >
            <h2>
              New card Submission Form
            </h2>
          </div>,
          <form
            className="new-card-form"
            onSubmit={[Function]}
          >
            <label
              htmlFor="text"
            >
              Your Message
            </label>
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              New card Submission Form
            </h2>,
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New card Submission Form",
            },
            "ref": null,
            "rendered": "New card Submission Form",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="text"
              >
                Your Message
              </label>,
              <textarea
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                value=""
              />,
              <label
                htmlFor="emoji"
              >
                 Emoji
              </label>,
              <input
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
                value=""
              />,
              <input
                className="new-card-form__form-button"
                type="submit"
                value="Submit Message"
              />,
            ],
            "className": "new-card-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Message",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Your Message",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "text",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Emoji",
                "htmlFor": "emoji",
              },
              "ref": null,
              "rendered": " Emoji",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-select",
                "name": "emoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-button",
                "type": "submit",
                "value": "Submit Message",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`NewCardForm that it matches an existing snapshot for delete 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    deleteCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="new-card-form__header"
        >
          <h2>
            New card Submission Form
          </h2>
        </div>,
        <form
          className="new-card-form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="text"
          >
            Your Message
          </label>
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            onChange={[Function]}
            value=""
          />
          <label
            htmlFor="emoji"
          >
             Emoji
          </label>
          <input
            className="new-card-form__form-select"
            name="emoji"
            onChange={[Function]}
            value=""
          />
          <input
            className="new-card-form__form-button"
            type="submit"
            value="Submit Message"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            New card Submission Form
          </h2>,
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New card Submission Form",
          },
          "ref": null,
          "rendered": "New card Submission Form",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="text"
            >
              Your Message
            </label>,
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />,
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>,
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />,
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />,
          ],
          "className": "new-card-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Message",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Your Message",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "text",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Emoji",
              "htmlFor": "emoji",
            },
            "ref": null,
            "rendered": " Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-button",
              "type": "submit",
              "value": "Submit Message",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="new-card-form__header"
          >
            <h2>
              New card Submission Form
            </h2>
          </div>,
          <form
            className="new-card-form"
            onSubmit={[Function]}
          >
            <label
              htmlFor="text"
            >
              Your Message
            </label>
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              New card Submission Form
            </h2>,
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New card Submission Form",
            },
            "ref": null,
            "rendered": "New card Submission Form",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="text"
              >
                Your Message
              </label>,
              <textarea
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                value=""
              />,
              <label
                htmlFor="emoji"
              >
                 Emoji
              </label>,
              <input
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
                value=""
              />,
              <input
                className="new-card-form__form-button"
                type="submit"
                value="Submit Message"
              />,
            ],
            "className": "new-card-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Message",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Your Message",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "text",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Emoji",
                "htmlFor": "emoji",
              },
              "ref": null,
              "rendered": " Emoji",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-select",
                "name": "emoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-button",
                "type": "submit",
                "value": "Submit Message",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`NewCardForm that it matches an existing snapshot for delete 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    deleteCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="new-card-form__header"
        >
          <h2>
            New card Submission Form
          </h2>
        </div>,
        <form
          className="new-card-form"
          onSubmit={[Function]}
        >
          <label
            htmlFor="text"
          >
            Your Message
          </label>
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            onChange={[Function]}
            value=""
          />
          <label
            htmlFor="emoji"
          >
             Emoji
          </label>
          <input
            className="new-card-form__form-select"
            name="emoji"
            onChange={[Function]}
            value=""
          />
          <input
            className="new-card-form__form-button"
            type="submit"
            value="Submit Message"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            New card Submission Form
          </h2>,
          "className": "new-card-form__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New card Submission Form",
          },
          "ref": null,
          "rendered": "New card Submission Form",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="text"
            >
              Your Message
            </label>,
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />,
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>,
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />,
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />,
          ],
          "className": "new-card-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Message",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Your Message",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "text",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Emoji",
              "htmlFor": "emoji",
            },
            "ref": null,
            "rendered": " Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-button",
              "type": "submit",
              "value": "Submit Message",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="new-card-form__header"
          >
            <h2>
              New card Submission Form
            </h2>
          </div>,
          <form
            className="new-card-form"
            onSubmit={[Function]}
          >
            <label
              htmlFor="text"
            >
              Your Message
            </label>
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              value=""
            />
            <label
              htmlFor="emoji"
            >
               Emoji
            </label>
            <input
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
            />
            <input
              className="new-card-form__form-button"
              type="submit"
              value="Submit Message"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              New card Submission Form
            </h2>,
            "className": "new-card-form__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New card Submission Form",
            },
            "ref": null,
            "rendered": "New card Submission Form",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="text"
              >
                Your Message
              </label>,
              <textarea
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                value=""
              />,
              <label
                htmlFor="emoji"
              >
                 Emoji
              </label>,
              <input
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
                value=""
              />,
              <input
                className="new-card-form__form-button"
                type="submit"
                value="Submit Message"
              />,
            ],
            "className": "new-card-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Message",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Your Message",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-textarea",
                "name": "text",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Emoji",
                "htmlFor": "emoji",
              },
              "ref": null,
              "rendered": " Emoji",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-select",
                "name": "emoji",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-card-form__form-button",
                "type": "submit",
                "value": "Submit Message",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
